tests/accesscontrol/helper/helper.go

```diff
@@ -156,3 +181,247 @@ func DefineAndCreateServiceOnCluster(name, namespace string, port int32, targetP
 
 	return nil
 }
+
+// Finds the MachineConfigPools that manage worker nodes of the cluster. Returns a map of MCP names
+// to slices of worker names.
+func GetWorkersMCPs() (mcpNodes map[string][]string, err error) {
+	// Get the eco-goinfra client.
+	egiClient := globalhelper.GetEcoGoinfraClient()
+	if egiClient == nil {
+		return nil, fmt.Errorf("eco-goinrfa client is not initialized")
+	}
+
+	nodeBuilders, err := egiNodes.List(egiClient, metav1.ListOptions{})
+	if err != nil {
+		return nil, fmt.Errorf("failed to get nodes: %w", err)
+	}
+
+	mcpBuilders, err := egiMco.ListMCP(egiClient, client.ListOptions{})
+	if err != nil {
+		return nil, fmt.Errorf("failed to list MachineConfigPools: %w", err)
+	}
+
+	mcps := map[string][]string{}
+
+	// For each mcp, add entry to the herlpe mcps map if it has worker nodes.
+	for i := range mcpBuilders {
+		mcpNodes := []string{}
+		mcp := mcpBuilders[i].Object
+
+		if mcp.Status.MachineCount == 0 {
+			continue
+		}
+
+		labelSelector, err := metav1.LabelSelectorAsSelector(mcp.Spec.NodeSelector)
+		if err != nil {
+			return nil, fmt.Errorf("failed to get labelselector from mcp %s: %w", mcp.Name, err)
+		}
+
+		for i := range nodeBuilders {
+			node := nodeBuilders[i].Object
+
+			nodeLabelsSet := labels.Set(node.Labels)
+			if !labelSelector.Matches(nodeLabelsSet) {
+				continue
+			}
+
+			if _, exist := node.Labels["node-role.kubernetes.io/worker"]; !exist {
+				break
+			}
+			mcpNodes = append(mcpNodes, node.Name)
+		}
+
+		// Continue if this MCP doesn't manage any worker node
+		if len(mcpNodes) == 0 {
+			glog.V(5).Infof("MCP=%s doesn't manage any worker node", mcp.Name)
+
+			continue
+		}
+
+		mcps[mcp.Name] = mcpNodes
+
+		glog.V(5).Infof("MCP=%s is candidate, num workers=%d", mcp.Name, len(mcpNodes))
+	}
+
+	return mcps, nil
+}
+
+// Gets the label set that a MachineConfig needs so it's used by a MachineConfigPool.
+//
+// The MachineConfigPool watches all the MachineConfig objects in the cluster but it only uses the ones that matches
+// the selector in MachineconfigPool.Spec.MachineConfigSelector to create the final rendered-xxxx MachineConfig.
+//
+// The code uses the MachineConfigSelector on each MachineConfig found and returns the labels of the MachineConfig that
+// matches that selector.
+func GetMachineConfigTargetLabels(mcpName string) (map[string]string, error) {
+	// Get the eco-goinfra client.
+	egiClient := globalhelper.GetEcoGoinfraClient()
+	if egiClient == nil {
+		return nil, fmt.Errorf("eco-goinrfa client is not initialized")
```

Comment:
`eco-goinrfa` --> `eco-goinfra`

Reply:
Thanks. Fixed in https://github.com/redhat-best-practices-for-k8s/certsuite-qe/pull/1059/commits/adee779dcdbe6f3b0536704cd7743230ebf8d04d